tortoise/backends/base/schema_generator.py

```diff
@@ -143,54 +144,48 @@ def _make_hash(*args: str, length: int) -> str:
         # Hash a set of string values and get a digest of the given length.
         return sha256(";".join(args).encode("utf-8")).hexdigest()[:length]
 
-    def _generate_index_name(
-        self, prefix: str, model: "type[Model] | str", field_names: list[str]
+    def _get_index_name(
+        self, prefix: str, model: type[Model] | str, field_names: Sequence[str]
     ) -> str:
         # NOTE: for compatibility, index name should not be longer than 30
         # characters (Oracle limit).
         # That's why we slice some of the strings here.
+        template = "{prefix}_{table}_{field}_{hash}"
         table_name = model if isinstance(model, str) else model._meta.db_table
-        index_name = "{}_{}_{}_{}".format(
-            prefix,
-            table_name[:11],
-            field_names[0][:7],
-            self._make_hash(table_name, *field_names, length=6),
+        hash_ = self._make_hash(table_name, *field_names, length=6)
+        return template.format(
```

Comment:
I think it would be much more readable if the template wasn't separated by multiple lines from actual formatting.